src/security/BlindPeerConnector.cc

```diff
@@ -86,3 +86,13 @@ Security::BlindPeerConnector::noteNegotiationDone(ErrorState *error)
     }
 }
 
+Security::BlindPeerConnector::BlindPeerConnector(HttpRequestPointer &aRequest,
+        const Comm::ConnectionPointer &aServerConn,
+        const AsyncCallback<EncryptorAnswer> &aCallback,
+        const AccessLogEntryPointer &alp,
+        const time_t timeout) :
```

Comment:
With @rouskov removal of the `const` we must keep the definition and declarations matching.
```suggestion
        time_t timeout) :
```

Reply:
> we must keep the definition and declarations matching

They do match: In C++, `const`  in `const time_t` in function declaration is meaningless garbage -- no declaration user can benefit from that `const` being there (that is why some compilers warn about it) while `const`  in `const time_t foo` in function definition is a meaningful const-correctness feature -- both humans and compilers can benefit from it (matching `time_t` in the declaration).

Please withdraw your change requests.

Reply:
FYI, One of the compilers that complain about "pointless" use of `const` is doxygen. Which complains as well when defintion/declaration are different `const`-ness.

Reply:
> FYI, One of the compilers that complain about "pointless" use of `const` is doxygen. Which complains as well when defintion/declaration are different `const`-ness.

If Doxygen complains about both pointless const in declarations AND a C++-allowed differences between definitions and declarations, then we have no better choice than to ignore those complaints because there is no way for us to satisfy Doxygen without losing const-correctness completely. 

BTW, there are dozens of these "different" declaration/definition cases in Squid already, and there will be more. Evidently, Doxygen can grok this code even if it (incorrectly) warns about it.

> Also, as you should know the reason being that when `const` is pointless, it became pointless because the variable value was _copied_ into the stack by `lvalue`. So the function/method does not have access to change the original caller variable even if it changes the variable it got. Even telling the compiler the variable is `const` is pointless - both caller and receiver already know it is `lvalue` passing.

I am not sure what the purpose of the above paragraph is, but the reason to add `const` to parameters in function definitions is to tell both the human reader and the compiler that the parameter cannot be modified by the function (including functions it calls). Many humans find it easier to read and refactor code when they know that a variable cannot be modified in the middle of it. Compilers make sure that invariant actually holds (and, in some rare complex cases, might even benefit from that `const` knowledge when optimizing code). In addition, using `const` in this context makes writing better code a bit easier -- fewer exceptions to worry about when following the "default to const" or "start with const" principle (that we should all follow to compensate for this C/C++ weakness IMHO).